docs/Platforms.md

```diff
@@ -18,6 +18,8 @@ Optionally, `msquic.dll` can be built with OpenSSL (see below for more details)
 
 On Linux, MsQuic relies on [OpenSSL](https://www.openssl.org/) for TLS 1.3 functionality.
 
+> **Important** MsQuic does not currently support OpenSSL 3.x. In particular, MsQuic is incompatible with Alpine 3.17+, Debian 12+, and Ubuntu 22.04+ since they exclusively offer OpenSSL 3.x. Recent versions of other distros may be similar.
```

Comment:
> MsQuic is incompatible with ...

This is not quite true. MsQuic supports statically linking all of openssl, which allows it to work on these releases. If you're using the inbox openssl, then yes, it won't work.

Reply:
One can of course also just build older OpenSSL or use binaries from older version or different distro. We did that at least in one case. So this should make it clear it is about distro provided binaries.

Reply:
It is entirely possible I don't understand. Does the statically linked version of OpenSSL play nicely with the one we use for HTTP 1 and 2 TLS scenarios?

Reply:
What about FIPS? I assume we should tell folks that we don't have a FIPS story for HTTP3. Fair?

Reply:
> Does the statically linked version of OpenSSL play nicely with the one we use for HTTP 1 and 2 TLS scenarios?

If MsQuic statically links OpenSSL, then it ends up using its own version/copy of OpenSSL that is different than the one uses .NET. Everything still works, but it's not the most efficient.

> What about FIPS? I assume we should tell folks that we don't have a FIPS story for HTTP3. Fair?

Statically linking breaks FIPS. The only way to get FIPS is to use the distro version.

Reply:
Does .NET use the statically-linked OpenSSL? I honestly don't know. I'm not on the networking team.

It sounds like the second commit on this PR is wrong: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/777. I didn't understand that MsQuic used static linking. Correct?

Reply:
No. We depend on OS libraries to avoid servicing and responsibility for security fixes. (and get FIPS stamp if that is any value) 

I feel there may be terminology mismatch between what MsQuic _can_ do and supports vs what is shipped as available and signed binaries. (easily consumable by .NET users) 

Last dimension is testing. If you look through the Helix images, on several platforms we did various hacks to get MsQuic in. For example we have Ubuntu 22 with Quic & OpenSSL 1.1 even if the vendor does not provide one. That for example allows us to test coexistence with OpenSSL 3 & 1.1.

Reply:
> get FIPS stamp if that is any value

Our customers have FIPS scenarios so definitely valuable.

Reply:
FIPS is a whole complex topic unto itself. IIRC, FIPS certification is not for OpenSSL as a whole, it's only done for the libssl portion that implements the crypto algorithms themselves. MsQuic relies on QuicTLS which is the QUIC capable fork of OpenSSL. QuicTLS doesn't change the libssl portion, so if a distro vendor or consumer has a FIPS certified version of libssl that can/will be loaded instead of the one included in QuicTLS.

Reply:
It is easy to test, @samsp-msft . Can you try running an app on Ubuntu 22.04+, Alpine 3.17+, or Debian 12+, per our generic guidance?

Reply:
Sorry, I'm heads down on diagnostics work right now. @wfurt - are you able to run the tests?

Reply:
I can do it if you point me to some samples.

Reply:
@richlander - the setup instructions and httpclient example code are all in this [doc](https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-http3). The doc also links to a cloudflare server to test with.

As the issue is TLS support, if the client works, the server should too, but to verify do a `dotnet new web`, and then use [this](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?view=aspnetcore-7.0#getting-started) modification to add an http3 endpoint on port 5001. As the client is forcing http/3, if that works for https://localhost:5001/, then you have proof both work.

Reply:
You can go to public HTTP/3 and avoid all the Kestrel setup...

https://github.com/dotnet/runtime/blob/d2a87fe8290e13bfc4d0901fc30595dfbd6bc7ac/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs#L1717-L1723